pkg/kubernetes-mcp-server/cmd/root.go

```diff
@@ -1,140 +1,181 @@
-			os.Exit(1)
-		}
-		listOutput := output.FromString(viper.GetString("list-output"))
-		if listOutput == nil {
-			fmt.Printf("Invalid output name: %s, valid names are: %s\n", viper.GetString("list-output"), strings.Join(output.Names, ", "))
-			os.Exit(1)
-		}
-		klog.V(1).Info("Starting kubernetes-mcp-server")
-		klog.V(1).Infof(" - Profile: %s", profile.GetName())
-		klog.V(1).Infof(" - ListOutput: %s", listOutput.GetName())
-		klog.V(1).Infof(" - Read-only mode: %t", viper.GetBool("read-only"))
-		klog.V(1).Infof(" - Disable destructive tools: %t", viper.GetBool("disable-destructive"))
-		if viper.GetBool("version") {
-			fmt.Println(version.Version)
-			return
-		}
-		mcpServer, err := mcp.NewSever(mcp.Configuration{
-			Profile:            profile,
-			ListOutput:         listOutput,
-			ReadOnly:           viper.GetBool("read-only"),
-			DisableDestructive: viper.GetBool("disable-destructive"),
-			Kubeconfig:         viper.GetString("kubeconfig"),
-		})
-		if err != nil {
-			fmt.Printf("Failed to initialize MCP server: %v\n", err)
-			os.Exit(1)
-		}
-		defer mcpServer.Close()
-
-		ssePort := viper.GetInt("sse-port")
-		if ssePort > 0 {
-			sseServer := mcpServer.ServeSse(viper.GetString("sse-base-url"))
-			defer func() { _ = sseServer.Shutdown(cmd.Context()) }()
-			klog.V(0).Infof("SSE server starting on port %d and path /sse", ssePort)
-			if err := sseServer.Start(fmt.Sprintf(":%d", ssePort)); err != nil {
-				klog.Errorf("Failed to start SSE server: %s", err)
-				return
-			}
-		}
+	genericiooptions.IOStreams
+}
+
+func NewMCPServerOptions(streams genericiooptions.IOStreams) *MCPServerOptions {
+	return &MCPServerOptions{
+		IOStreams:  streams,
+		Profile:    "full",
+		ListOutput: "table",
+	}
+}
 
-		httpPort := viper.GetInt("http-port")
-		if httpPort > 0 {
-			httpServer := mcpServer.ServeHTTP()
-			klog.V(0).Infof("Streaming HTTP server starting on port %d and path /mcp", httpPort)
-			if err := httpServer.Start(fmt.Sprintf(":%d", httpPort)); err != nil {
-				klog.Errorf("Failed to start streaming HTTP server: %s", err)
-				return
+func NewMCPServer(streams genericiooptions.IOStreams) *cobra.Command {
+	o := NewMCPServerOptions(streams)
+	cmd := &cobra.Command{
+		Use:     "kubernetes-mcp-server [command] [options]",
+		Short:   "Kubernetes Model Context Protocol (MCP) server",
+		Long:    long,
+		Example: examples,
+		RunE: func(c *cobra.Command, args []string) error {
+			if err := o.Complete(); err != nil {
+				return err
+			}
+			if err := o.Validate(); err != nil {
+				return err
+			}
+			if err := o.Run(); err != nil {
+				return err
 			}
-		}
 
-		if err := mcpServer.ServeStdio(); err != nil && !errors.Is(err, context.Canceled) {
-			panic(err)
-		}
-	},
+			return nil
+		},
+	}
+
+	cmd.Flags().BoolVar(&o.Version, "version", o.Version, "Print version information and quit")
+	cmd.Flags().IntVar(&o.LogLevel, "log-level", o.LogLevel, "Set the log level (from 0 to 9)")
```

Comment:
Maybe instead of having a separate `log-level` flag, we can directly wire the `-v` in klog.